### ISSUE TITLE
SecDefaultAction at phase3/4 needed to error log CRS triggered rules

### ISSUE THREAD

**Issue by M4tteoP** [corazawaf/coraza#494]
## Description
Differently from Modsecurity, in Coraza, in order to have phase 3 (to confirm) and phase 4 CRS rules logged in the error logs when triggered it is needed to add SecDefaultAction for these two phases.
Opening this issue in order to keep tracking this behaviour.
The next step should be to dive into the Modsecurity code to figure out the difference.
For more context: https://github.com/corazawaf/coraza/pull/483#discussion_r1008562465

**Comment by fzipi** [comment]
I did take a look and forgot. Let me do a second pass.

**Comment by jcchavezs** [comment]
Any movement on this @fzipi?

**Comment by fzipi** [comment]
Let me recap my findings from the source code and ancient history.

👉  `SecDefaultAction` is there just to have something as default if your rule doesn't have it.

The reality is that not even Trustwave liked it. See [number 2 here](https://www.trustwave.com/en-us/resources/blogs/spiderlabs-blog/three-modsecurity-rule-language-annoyances/). 😄 

Documentation says:
> Every rule following a previous SecDefaultAction directive in the same configuration context will inherit its settings unless more specific actions are used. Every SecDefaultAction directive must specify a disruptive action and a processing phase and cannot contain metadata actions.

ModSecurity source code doesn't log by default in phases 3-4 unless it is instructed to do so. So this is the same, I guess.

Some things I see in Coraza:
* [Here](https://github.com/corazawaf/coraza/blob/cc0c32509a39b0354a43fbca528058445fbcd33e/testing/engine/disruptive_actions.go#L307) we use the `logdata` (metadata action), and [we expect to succeed](https://github.com/corazawaf/coraza/blob/cc0c32509a39b0354a43fbca528058445fbcd33e/testing/engine/disruptive_actions.go#L273). So this must be changed to be compatible.
* MS source code has: "The transformation none is not suitable to be part of the SecDefaultActions". So we need to exclude `t:none` from a valid default action. In fact, v2 says in the CHANGES file "The use of tfns are deprecated in SecDefaultAction. Should be forbid in the future.".
* There can only be one `SecDefaultAction` per phase. We need to verify we check this.

In summary:
- if the rule doesn't have a log or auditlog, and there is no defaultaction with log or auditlog, there should be no logging.
- Coraza is doing exactly what we tell it to do.

👉  I will review the need for adding this by default in the CRS, because we add logging to each rule that needs it in phase 3 and 4, or we need to set up the default action 🤷 

BUT, we need to be very careful when logging in phase 3/4 because it might be **really** sensitive.

**Comment by dune73** [comment]
I second your reasoning @fzipi. Also what I had in mind about the default action and I do not like it either. Or at least not they way works in practice.

**Comment by M4tteoP** [comment]
All the comments related to Coraza should be addressed by https://github.com/corazawaf/coraza/pull/712.
In https://github.com/corazawaf/coraza/pull/684 I had to add explicitly `SecDefaultAction "phase:5,log,auditlog,pass"` to make CRS tests (Related to correlation rules) work. I'm still a bit surprised how they are not needed with ModSec, but I think that we are agreeing that Coraza is working as expected.
We might be able to close this one (after https://github.com/corazawaf/coraza/pull/712), and move the conversation CRS side about adding logging by default
